sirepo/auth/__init__.py

```diff
@@ -690,8 +702,7 @@ def _create_roles_for_new_user(self, method):
     def _create_user(self, module, want_login):
         u = simulation_db.user_create()
         if want_login:
-            self._login_user(module, u)
-            self._create_roles_for_new_user(module.AUTH_METHOD)
+            self._login_user(module, u, want_new_roles=True)
```

Comment:
I'm having a hard time understanding the second half of your comment. Is this what you mean?

```python3
    def _create_user(self, module, want_login):
        def _create_roles_for_new_user(self, method, uid):
            r = sirepo.auth_role.for_new_user(is_guest=method == METHOD_GUEST)
            if r:
                self.qcall.auth_db.model("UserRole").add_roles(roles=r, uid=uid)

        u = simulation_db.user_create()
        if want_login:
            self._login_user(module, u)
        self._create_roles_for_new_user(module.AUTH_METHOD, uid=u)
        return u
```

Reply:
```py
def _create_user(self, module, want_login):
     u = simulation_db.user_create()
     r = sirepo.auth_role.for_new_user(is_guest=method == METHOD_GUEST)
     if r:
         self.qcall.auth_db.model("UserRole").add_roles(roles=r, uid=uid)
     if want_login:
         self._login_user(module, u)
     return u
```

The want_login has to go at the end because there's check_role_user which needs to be added above.

Reply:
[634efd8](https://github.com/radiasoft/sirepo/pull/7070/commits/634efd840a1338c81fced52736fe4cfde315bb63)